internal/controller/types.go

```diff
@@ -35,15 +35,24 @@ type ProviderConfig struct {
 	URL string `json:"url,omitempty"`
 	// Path to the file containing API provider credentials in the app server container.
 	// default to "bam_api_key.txt"
-	CredentialsPath string `json:"credentials_path" default:"bam_api_key.txt"`
+	CredentialsPath string `json:"credentials_path,omitempty" default:"bam_api_key.txt"`
 	// List of models from the provider
 	Models []ModelConfig `json:"models,omitempty"`
 	// Provider type
 	Type string `json:"type,omitempty"`
-	// Azure deployment name
-	AzureDeploymentName string `json:"deployment_name,omitempty"`
 	// Watsonx Project ID
 	WatsonProjectID string `json:"project_id,omitempty"`
+	// Azure OpenAI Config
+	AzureOpenAIConfig *AzureOpenAIConfig `json:"azure_openai_config,omitempty"`
+}
+
+type AzureOpenAIConfig struct {
```

Comment:
Are other fields listed in [the JIRA ticket](https://issues.redhat.com/browse/OLS-620) in the azureconfig section not neccessary?
```yaml           
   azureconfig:
                url: http://someurl
                client_secret_path: <path_to_file>
                tenant_id_path: <path_to_tenant_id>
                client_id_path: <path_to_client_id>
                api_key_path: <path_to_api_key>  (alternative supported today)
```